### DISCUSSION TITLE
Secret Sync alpha release!

### DISCUSSION THREAD

**Discussion by ramizpolic** [bank-vaults/bank-vaults#2260] [Announcements]
## 🎉 New Project - Secret Sync 🎉
We are excited to announce the alpha release of [**Secret Sync**](https://github.com/bank-vaults/secret-sync), a project designed to _streamline secret synchronization_.

**Secret Sync** exposes a generic way to interact with external secret storage systems like HashiCorp Vault and provides a set of API models to interact and orchestrate the synchronization of secrets between them.

**🌟 Features**

* Simple integration with a variety of secret storage systems
* User-friendly API to facilitate interaction between different stores
* Pipeline-like syntax for defining synchronization actions on a secret level
* Advanced templating capabilities for transforming secrets

To get familiarized, find the [Getting Started](https://github.com/bank-vaults/secret-sync/blob/main/README.md#getting-started) guide which will walk you through how to use Secret Sync in a GitOps way.

❓ **What's next?**
We are actively working on consolidating our APIs and expanding the list of supported secret storage providers.
Check out our [Roadmap](https://github.com/bank-vaults/secret-sync/issues) to see the new features we are working on.

🗣️ **We are looking for feedback!**
Your insights are crucial to us. You can support us with your feedback, bug reports, and feature requests by responding to this [announcement](https://github.com/bank-vaults/secret-sync/discussions/61), or opening an [issue](https://github.com/bank-vaults/secret-sync/issues).

This release marks another important milestone for [Bank-Vaults](https://github.com/bank-vaults) in our mission to simplify and enhance secret management in the Cloud Native ecosystem.

**Comment by ernadh** [comment]
Congratulations! This is an important milestone!

Do we provide a mechanism to synchronize Kubernetes `Secret` resource with Vault?

**Reply by sagikazarmark** [reply]
Although we may add Kubernetes secret support at some point, for Vault to k8s Secret synchronization we recommend checking [External Secrets](https://external-secrets.io/latest/)

**Reply by ernadh** [reply]
Yeah, the motivation behind the question was mostly to be able to tell folks to phase out External Secrets as part of `secret-sync` adoption. :)

**Reply by sagikazarmark** [reply]
I think it depends on the use case, but the way I see it, ESO and BV complement each other: ESO works well when you go all in on Kubernetes secrets (which is your decision but brings a number of issues of its own).

BV is the way to go when you don't want to use k8s secrets, but don't want to hardcode Vault into your application either. BV is the service mesh of secret management. :P

Some buffoon actually did a talk on this: https://presentations.cncf.io/presentations/automate-secret-rotation-in-kubernetes-then-get-out-of-the-way/

**Reply by ernadh** [reply]
You got me with BV being the service mesh of secret management! Nice talk too!
